src/main/java/com/mongodb/jdbc/MongoResultSet.java

```diff
@@ -805,35 +743,26 @@ public String getCursorName() throws SQLException {
     @Override
     public ResultSetMetaData getMetaData() throws SQLException {
         checkClosed();
-        checkAndCacheFirstDocAndMetaData();
-        if (current != null) {
-            rsMetaData = new MongoResultSetMetaData(current);
-        }
-        return rsMetaData;
```

Comment:
we've been able to remove a good amount of edge-case handling around the first doc and empty result sets now that empty result sets are no longer a special case

Reply:
This part makes me very happy. The handling between empty and one result was gross.